src/main/java/com/bobocode/creditadvisoryapp/model/Advisor.java

```diff
@@ -0,0 +1,65 @@
+package com.bobocode.creditadvisoryapp.model;
+
+import com.bobocode.creditadvisoryapp.enums.AppStatus;
+import com.bobocode.creditadvisoryapp.enums.Role;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Entity
+@Table(name = "advisors")
+public class Advisor extends User {
+
+  @Enumerated(EnumType.STRING)
+  private Role role;
+
+  @OneToMany(mappedBy = "advisor", cascade = {CascadeType.PERSIST, CascadeType.REMOVE}, orphanRemoval = true)
+  @ToString.Exclude
+  @Setter(AccessLevel.PRIVATE)
+  private List<Application> apps = new ArrayList<>();
+
+  public void addApplication(Application app) {
+    apps.add(app);
+    app.setAdvisor(this);
+  }
+
+  public void removeApp(Application app) {
+    apps.remove(app);
+    app.setAdvisor(null);
+  }
+
+  public void validateAssignment() {
+    var isAlreadyAssigned = apps.stream()
+        .map(Application::getStatus)
+        .filter(Objects::nonNull)
```

Comment:
You don't need this filtering as null != AppStatus.ASSIGNED, so if Status is null it just will return false

Reply:
But in general, I would add a default status - `NEW` for each newly created application